pkg/transparentproxy/config/config.go

```diff
@@ -89,6 +125,98 @@ func NewValueOrRangeList[T ~[]uint16 | ~uint16 | ~string](v T) ValueOrRangeList
 	}
 }
 
+var _ json.Unmarshaler = (*Port)(nil)
+
+type Port uint16
+
+func (p *Port) String() string { return strconv.Itoa(int(*p)) }
+
+func (p *Port) Type() string { return "uint16" }
+
+func (p *Port) Set(s string) error {
+	var err error
+
+	if *p, err = parsePort(s); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Port) UnmarshalJSON(bs []byte) error { return p.Set(string(bs)) }
+
+var _ json.Unmarshaler = &Ports{}
+
+type Ports []Port
+
+func (p *Ports) String() string {
+	var ports []string
+	for _, port := range *p {
+		ports = append(ports, strconv.Itoa(int(port)))
+	}
+	return strings.Join(ports, ",")
+}
+
+func (p *Ports) Type() string { return "uint16[,...]" }
+
+func (p *Ports) Set(s string) error {
+	*p = nil
+
+	if s = strings.TrimSpace(s); s == "" {
+		return nil
+	}
+
+	for _, port := range strings.Split(s, ",") {
+		trimmedPort := strings.TrimSpace(port)
+		if trimmedPort == "" {
+			continue
+		}
+
+		parsedPort, err := parsePort(trimmedPort)
+		if err != nil {
+			return err
+		}
+
+		*p = append(*p, parsedPort)
+	}
+
+	return nil
+}
+
+func parsePort(s string) (Port, error) {
+	u, err := parseUint16(s)
+
+	if err != nil || u == 0 {
```

Comment:
Do we support specifying a `0` value to disable something?

Reply:
No, I don't think we should as it would duplicate the existing logic and is unnecessary